include/klee/Expr/Assignment.h

```diff
@@ -102,17 +104,22 @@ inline ref<Expr> Assignment::evaluate(const Array *array,
   }
 }
 
-inline ref<Expr> Assignment::evaluate(ref<Expr> e) const {
-  AssignmentEvaluator v(*this);
+inline ref<Expr> Assignment::evaluate(ref<Expr> e, bool allowFreeValues) const {
+  AssignmentEvaluator v(*this, allowFreeValues);
   return v.visit(e);
 }
 
 template <typename InputIterator>
-inline bool Assignment::satisfies(InputIterator begin, InputIterator end) {
-  AssignmentEvaluator v(*this);
-  for (; begin != end; ++begin)
-    if (!v.visit(*begin)->isTrue())
+inline bool Assignment::satisfies(InputIterator begin, InputIterator end,
+                                  bool allowFreeValues) {
+  AssignmentEvaluator v(*this, allowFreeValues);
+  for (; begin != end; ++begin) {
+    if ((*begin)->getWidth() == Expr::Bool && !v.visit(*begin)->isTrue())
```

Comment:
Please separate the logic for processing non-boolean expressions into a separate function